openapi2kong/jsonschema.go

```diff
@@ -51,10 +51,12 @@ func dereferenceSchema(sr *base.SchemaProxy, seenBefore map[string]*base.SchemaP
 
 // extractSchema will extract a schema, including all sub-schemas/references and
 // return it as a single JSONschema string. All components will be moved under the
-// "#/definitions/" key.
-func extractSchema(s *base.SchemaProxy) string {
+// "#/definitions/" key. Along with that, it will also return the schema as a
+// map, so that if any further operations or validations need to be done on the schema
+// they could be performed.
+func extractSchema(s *base.SchemaProxy) (string, map[string]interface{}) {
```

Comment:
In this function we already loop over all references. Can't we return a bool here that indicates whether the schema dereferenced has a proper `type`? Instead of introducing a new function `fetchTopLevelType` below, which does all the looping again?

Reply:
We may be able to do that but I like it being done in a separate function to keep the two different concerns separated. `openapi2kong` code is pretty huge and mixing more unrelated things in existing functions can hamper with understandability for future contributions.

Reply:
I was thinking to integrate it here, for the exact same reason; maintainability.